tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -431,7 +431,10 @@ void test_mem_part_assert_add_overmax(void)
 	/* Add one more partition will trigger assert due to exceeding */
 	k_mem_domain_add_partition(&test_domain, &exceed_part);
 
-	/* should not reach here */
+	/* should not reach here but if it does clean up flag before
+	 * indicating failure
+	 */
+	need_recover_spinlock = false;
```

Comment:
This indeed should be added to clear that flag when it did not trigger the exception, I missed it. thank you!

Reply:
I agree this is needed but the overall PR is not needed as the bug in question has been fixed by a previous change.